levels/sre.tex

```diff
@@ -0,0 +1,368 @@
+    Certified Solutions Architect.
+    \\ [13em]
+\hline
+    % Level
+    4
+    &
+
+    % Product
+    \cellcolor{blue!25}
+    Leads the implementation of the isolated feature/improvement that
+    measurably and significantly impacts business outcomes from gathering
+    requirements to getting to the market stage.
+    &
+
+    % Communication
+    Writes technical articles/blog posts, delivers tech and lightning talks
+    representing the company's technical vision.
+    &
+
+    % Networking
+    Can write (or heavily modify) network infrastructure tooling (like Cilium)
+    to provide service level load balancing, multi-region connectivity, and
+    observability.
+    &
+
+    % Security
+    Writes technical articles on security aspects of the system, implements
+    significant security product innovations in the area delivered to customers.
+
+    \bigbreak
+
+    Understands and can apply advanced cryptographic principles. Understands
+    hashing (including for anonymization), when to use symmetric and asymmetric
+    cryptography, cipher modes, and TLS versions.
+
+    \bigbreak
+
+    Understands and can apply advanced network security principles. Understands
+    data extrusion prevention. Understands DDoS mitigation. Understands how to
+    monitor systems for rootkits and can deploy mitigation strategies when a system
+    is under attack.
+    &
+
+    % Systems Engineering
+    Can write data-race and deadlock free code. Can implement safe and
+    concurrent and/or parallel systems using minimum amount of shared state,
+    granular locking - systems that are easy to read, extend and troubleshoot.
+    &
+
+    % SRE
+    Can build and operate large scale, stable, and reliable production platform
+    environments like Kubernetes.
+
+    \bigbreak
+
+    Not expected to find any faults or provide any further suggestions in the
+    design document submitted by an Level 4 engineer.
+    \\ [13em]
+\hline
+    % Level
+    5
+    &
+
+    % Product
+    \cellcolor{blue!25}
+    Leads the implementation of a new product line or significant part of the
+    product to deliver it to the market in collaboration with all other teams.
+    &
+
+    % Communication
+    Writes advanced technical articles/blog posts, gaining significant industry
+    traction or delivers technical talks on major conferences representing the
+    company's vision.
+    &
+
+    % Networking
+    &
+
+    % Security
+    Researches and designs new security systems and protocols.
+    &
+
+    % Systems Engineering
+    Can implement production grade systems leveraging advanced low-level and/or
+    novel components like eBPF, control groups, or Noise Protocol Framework.
+    &
+
+    % SRE
+    Understands and use use advanced system performance troubleshooting
+    techniques like ptrace, strace, flamegraphs, or writing custom bpftrace
+    programs.
+
+    \bigbreak
+
+    Can deploy and operate databases at large scale. Understands index
```

Comment:
I'd move "basic" database stuff, like "can operate databases in production" to lvl 4. And maybe even failover/replication, i.e. "ability to configure and troubleshoot replication for failover and horizontal scaling purposes".